---
- name: Portainer container
  community.general.docker_container:
    name: portainer
    image: portainer/portainer-ce:latest
    hostname: portainer
    state: started
    pull: yes
    restart_policy: on-failure
    restart_retries: 2
    cpu_shares: 128
    volumes:
      - "/var/run/docker.sock:/var/run/docker.sock"
      - "/var/portainer:/data"
    purge_networks: yes
    networks:
      - name: p2p
    labels:
      traefik.enable: "true"
      traefik.docker.network: "p2p"
      # service
      traefik.http.services.portainer.loadbalancer.server.port: "9000"
      # http router
      traefik.http.routers.portainer-http.rule: "Host(`portainer.{{ domain }}`)"
      traefik.http.routers.portainer-http.entrypoints: "web"
      traefik.http.routers.portainer-http.service: "portainer"
      # https router
      traefik.http.routers.portainer-https.rule: "Host(`portainer.{{ domain }}`)"
      traefik.http.routers.portainer-https.entrypoints: "websecure"
      traefik.http.routers.portainer-https.service: "portainer"
      traefik.http.routers.portainer-https.tls: "true"
      traefik.http.routers.portainer-https.tls.certresolver: "cloudflare"
